File: ruby/leap/leap.rb
module Year
  def self.leap?(year)
    year.divisible_by?(4) &&
      !year.divisible_by?(100) ||
      year.divisible_by?(400)
  end
end

class Integer
  def divisible_by?(number)
    (self % number).zero?
  end
end

File: typescript/leap/leap.ts
export default function isLeapYear(year: number) {
  if (isMultipleOf(year, 100) && !isMultipleOf(year, 400)) {
    return false
  }
  return isMultipleOf(year, 4)
}

function isMultipleOf(n: number, factor: number) {
  return n % factor === 0
}
